### PR TITLE
Feature/733 738 - ui.config project

### PR DESCRIPTION
**Pull request by davidjgonzalez**
Folded in #733 which included Maven updates, as this was required to resolve some build issues.

@kwin trying to add you as reviewer.

The main changes in this PR are:

1. Broke out OSGi configurations into a ui.config container package project
2. Updated Maven plugins per #733 
3. Moved ACLs out to RepoInit scripts since the updates from #733 complained about the ACLs (when acHandling was set to merge, it complained there were no ACL lists, when this was removed, it complained there were ACL lists... maybe a bug?). Repoinit is preferrable anyhow, so i opted for that rather than debugging the other issue.

## Related Issue

#733 

## Motivation and Context

Adobe/Jackrabbit best practices

## How Has This Been Tested?

TBD - Will deploy to fresh AEM CS after a Draft review

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: ui.config/pom.xml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+
+    <!-- ====================================================================== -->
+    <!-- D E P E N D E N C I E S                                                -->
+    <!-- ====================================================================== -->
+    <dependencies>
+        <dependency>
```
Comment (kwin) [inline ui.config/pom.xml:83]: this dependency can be removed

File: ui.config/pom.xml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+                            <artifactId>assetshare.ui.apps.structure</artifactId>
+                        </repositoryStructurePackage>
+                    </repositoryStructurePackages>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>com.day.jcr.vault</groupId>
+                <artifactId>content-package-maven-plugin</artifactId>
+                <extensions>true</extensions>
+                <configuration>
```
Comment (kwin) [inline ui.config/pom.xml:71]: rather move this configuration to the parent pom

File: ui.config/src/main/content/META-INF/vault/definition/.content.xml [hunk 3]
```diff
@@ -0,0 +1,42 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (kwin) [inline ui.config/src/main/content/META-INF/vault/definition/.content.xml:1]: definitions don't need to be part of the package (redundant information, compare with https://github.com/apache/jackrabbit-filevault/blob/master/src/site/markdown/packagedefinition.md)

**Review by kwin — CHANGES_REQUESTED** [review]

**Comment by kwin** [comment]
> Moved ACLs out to RepoInit scripts since the updates from https://github.com/adobe/asset-share-commons/pull/733 complained about the ACLs (when acHandling was set to merge, it complained there were no ACL lists, when this was removed, it complained there were ACL lists... maybe a bug?)

I cannot reproduce. Filevault complained about unused `accessControlHandling` in module `ui.apps` (which is IMHO correct as that module doesn't contain any ACL lists).

File: ui.content/pom.xml [hunk 4]
```diff
@@ -52,7 +52,7 @@
                     <finalName>asset-share-commons.ui.content-${project.version}</finalName>
                     <packageType>content</packageType>
                     <allowIndexDefinitions>true</allowIndexDefinitions>
-                    <accessControlHandling>merge</accessControlHandling>
+                    <!--<accessControlHandling>merge</accessControlHandling>-->
```
Comment (kwin) [inline ui.content/pom.xml:55]: just remove, as it no longer contains ACLs

**Comment by kwin** [comment]
@davidjgonzalez Why were the reverts necessary?

**Comment by davidjgonzalez** [comment]
@kwin the problem was I needed to create the `/conf/.../policies[cq:Page]` node in repo init to permissions it - but then the `ui.content` package couldn't deploy the `nt:unstructured` nodes under that in merge mode.

**Comment by davidjgonzalez** [comment]
Validated against latest AEM CS .

**Pull request merged**